### ISSUE TITLE
[clang-tidy] readability-function-size.VariableThreshold not accept value -1

### ISSUE THREAD

**Issue by alex-brs** [llvm/llvm-project#58325]
clang-tidy v14 supports `readability-function-size.VariableThreshold` with value `-1` as its default value according to the documentation: 

https://releases.llvm.org/14.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/readability-function-size.html

However actually using `-1` results in

```console
error: invalid configuration value '-1' for option 'readability-function-size.VariableThreshold'; expected an integer
```

Unfortunately I did not have time to check in the latest version (I attempted using https://godbolt.org/ with clang-tidy as an add-on tool but could not get it to work).

**Comment by PiotrZSL** [comment]
This is because unsigned int is used for configuration parameter, and it can be negative, and at the same time check uses -1U as default. Having -1 basically overflow to max value and that way configuration is disabled. If you use --dump-config, it actually dump default limit as 4294967295. For me this is just a case where documentation should be updated.

**Comment by JoyNyuguto** [comment]
I suggest trying a different value for readability-function-size.VariableThreshold to see if it resolves the issue.Additionally, updating to the latest version of clang-tidy may also help resolve any issues with the default values.

@SamTebbs33 
Joy Nyuguto Outreachy

**Comment by PiotrZSL** [comment]
Fixed by 7a73da4c85a12341752a4573c55ebff46ba20db0
